kubernetes/cray-service/templates/deployment.yaml

```diff
@@ -36,7 +36,7 @@ spec:
 {{- if .Values.strategy }}
   strategy:
     {{ toYaml .Values.strategy | nindent 4 }}
-    {{- if eq (.Values.strategy.type | lower) "recreate" }}
+    {{- if and (eq (.Values.strategy.type | lower) "recreate") (not (hasKey .Values.strategy "rollingUpdate")) }}
```

Comment:
Current implementation was causing duplicate `strategy.rollingUpdate` lines, if `.Values.startegy.type` is set to `Recreate` and `.Values.strategy.rollingUpdate` is explicitly set to `null`.

Reply:
Think this leftover rollingUpdate: null should be removed entirely, that comment is out of date by 4ish years.

```
$ helm template . --set strategy.type=Recreate --set strategy.rollingUpdate=null                                                                                                                                                                                                      ~/src/wrk/github.com/Cray-HPE/base-charts/kubernetes/cray-service wmb
---
# Source: cray-service/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: release-name-cray-service
  labels:
    app.kubernetes.io/name: cray-service
    helm.sh/base-chart: cray-service-10.0.3
    helm.sh/chart: unknown-chart-name-unknown-chart-version
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
    
  annotations:
    cray.io/service: cray-service
    
spec:
  type: ClusterIP
  ports:
    - port: 80
      targetPort: http
      name: http
      protocol: TCP
  selector:
    app.kubernetes.io/name: cray-service
    app.kubernetes.io/instance: release-name
---
# Source: cray-service/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: release-name-cray-service
  labels:
    app.kubernetes.io/name: cray-service
    helm.sh/base-chart: cray-service-10.0.3
    helm.sh/chart: unknown-chart-name-unknown-chart-version
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
    
  annotations:
    cray.io/service: cray-service
    
spec:
  replicas: 1
  strategy:
    
    rollingUpdate: null
    type: Recreate
  selector:
    matchLabels:
      app.kubernetes.io/name: cray-service
      app.kubernetes.io/instance: release-name
  template:
    metadata:
      labels:
        app.kubernetes.io/name: cray-service
        app.kubernetes.io/instance: release-name
        
      annotations:
        service.cray.io/public: "true"
        
    spec:
      
      containers:
      
      
      
      
      volumes:
```

The only reason to have rollingUpdate: null was a helm 2 broken merge strategy:
https://github.com/helm/helm/issues/5144

I don't think we should retain it. I'll have to test this out quick but this reeks a bit of helm2 brokenness.